source/exact-match.d.ts

```diff
@@ -0,0 +1,55 @@
+/**
+Returns a `boolean` type to determine if the two types match exactly.
+
+This can be useful if another type wants to make a decision based on whether one type is exactly the same as the other, with the same keys of the same types.
```

Comment:
It would be nice to elaborate more here on specific real-world situation where it could come in handy.